### PR TITLE
Fixes ordering being inverted in queries that use map helpers.

### PR DESCRIPTION
**Pull request by telser**
This is to address #124 

Tests added are adapted from the `WhereCondition` test module for expediency.

### CODE REVIEW

File: orville-postgresql/src/Data/Map/Helpers.hs [hunk 1]
```diff
@@ -16,7 +16,7 @@ groupBy keyFunc = groupBy' mkEntry
     mkEntry a = (keyFunc a, a)
 
 groupBy' :: Ord k => (a -> (k, v)) -> [a] -> Map.Map k [v]
-groupBy' mkEntry as = Map.fromListWith (++) (map mkListEntry as)
+groupBy' mkEntry as = Map.fromListWith (flip (++)) (map mkListEntry as)
```
Comment (aaronallen8455) [inline orville-postgresql/src/Data/Map/Helpers.hs:19]: Can we use `DList` (or equivalent) instead to avoid left associating `++`?

Comment (telser) [reply orville-postgresql/src/Data/Map/Helpers.hs:19]: We could round trip through a `DList` I suppose, and exploring something like `DList` more could be great for the libpq version, but I certainly don't want to introduce a breaking api change with a bug fix like this.

Comment (aaronallen8455) [reply orville-postgresql/src/Data/Map/Helpers.hs:19]: Yes, round tripping is what I mean, no api change

Comment (qxjit) [reply orville-postgresql/src/Data/Map/Helpers.hs:19]: Agree that we avoid flipping `++` and putting the long list on the left side of it.  Round-tripping through `DList` seems like a reasonable and quick thing to do though.

**Review by aaronallen8455 — APPROVED** [review]

**Review by qxjit — APPROVED** [review]

**Pull request merged**